controllers/replace_failed_process_groups.go

```diff
@@ -76,13 +85,26 @@ func chooseNewRemovals(cluster *fdbtypes.FoundationDBCluster) bool {
 		needsReplacement, missingTime := processGroupStatus.NeedsReplacement(*cluster.Spec.AutomationOptions.Replacements.FailureDetectionTimeSeconds)
 		if needsReplacement && *cluster.Spec.AutomationOptions.Replacements.Enabled {
 			if len(processGroupStatus.Addresses) == 0 {
+				// Only replace process groups without an address if the cluster has the desired fault tolerance
+				// and is available.
+				hasDesiredFaultTolerance, err := internal.HasDesiredFaultTolerance(adminClient, cluster)
+				if err != nil {
+					log.Error(err, "Could not fetch if cluster has desired fault tolerance")
+					continue
+				}
+
+				if !hasDesiredFaultTolerance {
+					log.Info(
+						"Replace instance with missing address",
```

Comment:
If I'm interpreting the code correctly, in this branch we're choosing not to replace the instance because of the fault tolerance issues. Can we make the log message more descriptive so that's clear to the users?

Reply:
Yes, thanks for the catch 👍